accounting/bank/migrations/0004_auto_20180104_1124.py

```diff
@@ -2,12 +2,36 @@
 # Generated by Django 1.11.8 on 2018-01-04 11:24
 from __future__ import unicode_literals
 
+from django.conf import settings
 from django.db import migrations
 import django.contrib.postgres.fields.jsonb
 
 import accounting.bank.models
 
 
+def update_identification_keys(apps, schema_editor):
+    """
+    Updates any identification keys for all existing bank accounts.
+
+    Update: `accountNumber` -> `account_number`.
+    Add: `currency`.
+    Delete: All old keys.
+    """
+    BankAccount = apps.get_model('bank', 'BankAccount')
+    for bank_account in BankAccount.objects.all():
+        # Update: `accountNumber` -> `account_number`.
+        if 'accountNumber' in bank_account.identification:
+            bank_account.identification['account_number'] = bank_account.identification['accountNumber']
+        # Add: `currency`.
+        if 'currency' not in bank_account.identification:
+            bank_account.identification['currency'] = settings.DEFAULT_CURRENCY
```

Comment:
@antoviaque Ah, that's a good catch, when I was writing this I was thinking it was for the source currency, but yeah it's the target and wouldn't always be the correct default (and it's impossible to actually choose the correct default, except by querying TransferWise).

But yes this is because the CSV requires a target currency. The one that you generate from their UI already fills this part in for you, so that's presumably why you were seeing a locked-out target currency when trying to make a payment, since they just fetch it for you there too.

I just tested it out though, using their template, but making `targetCurrency` as `EUR` for everyone, and got this:

![selection_003](https://user-images.githubusercontent.com/10018065/35010671-745f3d4e-fb25-11e7-9295-c8108513cfbb.jpg)

So we should be good :+1:

Reply:
@UmanShahzad Thanks for checking this - and sounds good for the approach. CC @aliciaopencraft17